### PR TITLE
Minify JS for production

### LINKED ISSUE #861: Minify the JS  ( production )
####     Refer    #837 

Currently website use non-minifed JS file for serving
instead of it , we should use minified JS to decrease the size 
and for increasing the performance

### PR DESCRIPTION
**Pull request by HarshKhandeparkar**
Fixes #861 

This adds a new task to grunt which uglifies the dist and ui files for production and saves them under `image-sequencer.js` and `image-sequencer-ui.js` so that they are included in the index file but are minified for production

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by HarshKhandeparkar** [comment]
@jywarren this uses minified dist files for production instead of normal ones through a grunt task named `production`. Step \#2 towards gitignoring dist files.
cc @publiclab/is-reviewers

**Comment by HarshKhandeparkar** [comment]
This is instantly merge-able as it only changes the Gruntfile

File: Gruntfile.js [hunk 1]
```diff
@@ -28,6 +28,14 @@ module.exports = function(grunt) {
       ui: {
         src: ["examples/demo.js"],
         dest: "dist/image-sequencer-ui.js"
+      },
+      prodcore: {
```
Comment (jywarren) [inline Gruntfile.js]: whoa, interesting! What's prodcore?

Comment (HarshKhandeparkar) [reply Gruntfile.js]: Production, core src

Comment (HarshKhandeparkar) [reply Gruntfile.js]: Nothing big lol 😂

**Comment by jywarren** [comment]
ok! can you explain a bit more here the role of the .brow.js files? Thanks!

**Comment by HarshKhandeparkar** [comment]
They are just browserified files which are then uglified

**Comment by jywarren** [comment]
ah, but then can't we skip the ".brow"? I've seen it's more or less a
convention to just do library.js and library.min.js -- what do yo uthink?

**Comment by HarshKhandeparkar** [comment]
This is done because the minified file s are named unnder image-sequencer.js so the html files don't need to be changed. I did not know what would have happened if a file was uglified under the same name so i created the .brow.js files

**Comment by HarshKhandeparkar** [comment]
The .brow.js files are not included anywhere. They are just intermediate files for creating a minified library.js (without .min) which is what the html files include.

**Comment by HarshKhandeparkar** [comment]
Since these changes will only ever be made to gh-pages, the .brow.js will never be noticed

**Comment by jywarren** [comment]
oh i see. OK! Can you request a review from a few others just to double
check this is right? I'm not as familiar with this workflow. Thank you!

**Comment by HarshKhandeparkar** [comment]
Ok sure!

**Comment by HarshKhandeparkar** [comment]
@subhahu123 can you please review thi

**Comment by subhahu123** [comment]
Use .min extension for minified versions like
Image-sequencer.min.js

**Comment by HarshKhandeparkar** [comment]
.min extension is not used because the normal .js is included in the index file and changing the index file would be difficult. Since minified files without .min extension will only be pushed to gh-pages, it will create no difference

**Comment by HarshKhandeparkar** [comment]
@publiclab/is-reviewers please review.

**Comment by HarshKhandeparkar** [comment]
@tech4GT @Divy123 can you please review this?
cc @jywarren

**Comment by HarshKhandeparkar** [comment]
@jywarren are any changes reqd?

**Comment by jywarren** [comment]
OK, i read through carefully and i think we just need a README addition to explain the different commands briefly to people. How does that sound? I love what you've done here -- `grunt compile` vs. `grunt build` vs. `grunt production` -- i think they just need a little more explanation. Also, is there a way to add a description field to the Gruntfile tasks themselves?

Thanks @HarshKhandeparkar !!!

**Comment by HarshKhandeparkar** [comment]
Ok i will add it to readme tonight (IST) around 12am tonight in india so around 4hrs left.

**Comment by jywarren** [comment]
no rush, thank you!

**Comment by HarshKhandeparkar** [comment]
Shall i add the desc to contributing file instead of readme?

**Comment by jywarren** [comment]
Sure!

**Comment by HarshKhandeparkar** [comment]
@jywarren done with the documentation.
https://github.com/publiclab/image-sequencer/blob/prodify/CONTRIBUTING.md#grunt-tasks
Please let me know if any more changes are reqd. I will only be available for another 1/2 hour or 1hour max as it is already 12.30am here. Please let me know as soon as possible so that I can make the changes now itself. Thanks.

**Comment by HarshKhandeparkar** [comment]
@jywarren please review.

**Comment by jywarren** [comment]
This is super. We may even want to feature it more prominently!

**Pull request merged**